README.md

```diff
@@ -187,6 +187,20 @@ GitHub github = Feign.builder()
                      .target(GitHub.class, "https://api.github.com");
 ```
 
+If you need to pre-process the response before give it to the Decoder, you can use the [InterceptorDecoder](/feign/codec/InterceptorDecoder.java).
+
+```java
+GitHub github = Feign.builder()
+                     .decoder(
+                         new InterceptorDecoder(
```

Comment:
tempted to make this stuff private and only expose the builder..

Ex. mapAndDecode((response, type) -> myTransformingLogic(response, type), new GsonDecoder())

thoughts?